Lib/importlib/metadata.py

```diff
@@ -0,0 +1,271 @@
+import os
+import re
+import abc
+import sys
+import email
+import itertools
+import contextlib
+
+from configparser import ConfigParser
+from importlib import import_module
+from importlib.resources import Package
+from pathlib import Path
+from types import ModuleType
+from typing import Callable
+from zipfile import ZipFile
+
+
+__all__ = [
+    'Distribution',
+    'PackageNotFoundError',
+    'distribution',
+    'entry_points',
+    'resolve',
+    'version',
+    ]
+
+
+class PackageNotFoundError(ModuleNotFoundError):
+    """The package's metadata was not found."""
+
+
+class Distribution:
+    """A Python distribution package."""
+
+    @abc.abstractmethod
+    def load_metadata(self, name):
+        """Attempt to load metadata given by the name.
+
+        :param name: The name of the distribution package.
```

Comment:
I think this parameter description is a mistake. `name` is not the name of the distribution package, but the filename in the metadata. So in `requests-1.0.dist-info/METADATA`, `METADATA` is the name. Other calls might be looking for other metadata, like `load_metadata('entry_points.txt')` or `load_metadata('requires.txt')`. I see now in importlib_metadata, the docstring is a mismatch also, so I missed that in earlier revisions. I'll look into updating that.

Reply:
This has been corrected upstream.